pennylane/transforms/dynamic_one_shot.py

```diff
@@ -248,18 +248,25 @@ def measurement_with_no_shots(measurement):
 
     interface = interface or qml.math.get_deep_interface(circuit.data)
     interface = "numpy" if interface == "builtins" else interface
+    interface = "tensorflow" if interface == "tf" else interface
     active_qjit = qml.compiler.active()
 
     all_mcms = [op for op in aux_tapes[0].operations if is_mcm(op)]
     n_mcms = len(all_mcms)
-    mcm_samples = qml.math.hstack(tuple(res.reshape((-1, 1)) for res in results[-n_mcms:]))
+    mcm_samples = qml.math.hstack(
+        tuple(qml.math.reshape(res, (-1, 1)) for res in results[-n_mcms:])
+    )
     mcm_samples = qml.math.array(mcm_samples, like=interface)
     # Can't use boolean dtype array with tf, hence why conditionally setting items to 0 or 1
     has_postselect = qml.math.array(
-        [[int(op.postselect is not None) for op in all_mcms]], like=interface
+        [[int(op.postselect is not None) for op in all_mcms]],
+        like=interface,
```

Comment:
I don't think we want to convert back to interface data here.  If we call `dynamic_one_shot` after we've converted to numpy, we don't want to convert back into interface data.

Right now it looks like we are still calling `convert_to_numpy_parameters` after device preprocessing, but I could very easily see that changing in the future.

I was under the assumption that `interface` was passed to this function for the purpose of determining  jit-specific postselection handling.

Reply:
Maybe we could have a "postselection interface" and "result data interface"?

Reply:
> I was under the assumption that interface was passed to this function for the purpose of determining jit-specific postselection handling.

Yes, for pennylane that is the case, so I can definitely spend time investigating if we really need to convert to interface data, but for Catalyst we need to propagate jax data correctly.